t/lib/Mocker/Streaming.pm

```diff
@@ -0,0 +1,47 @@
+# This program is open source, licensed under the PostgreSQL License.
+# For license terms, see the LICENSE file.
+#
+# Copyright (C) 2012-2021: Open PostgreSQL Monitoring Development Group
+
+package Streaming;
+
+use strict;
+use warnings;
+
+# This module is a simple wrapper around the "query" sub existing in
+# check_pgactivity script. Its purpose is to capture and edit query results
+# to test some part of the streaming_delta check.
+#
+# You must load it on check_pgactivity execution using eg.:
+#   perl -It/ -MMocker::Streaming check_pgactivity --service streaming_delta
+
+CHECK {
+    # keep reference to old query sub
+    $main::{'query_orig'} = $main::{'query'};
+
+    # FIXME: check given query
+
+    # install wrapper around query sub to capture and modify the result
+    $main::{'query'} = sub {
+        my $res;
+
+        $res = $main::{'query_orig'}->(@_);
+
+        return $res unless $_[1] =~ m/FROM pg_stat_replication/;
+
+        # mock 1MB of write delta.
```

Comment:
I'm fine with this approach (especially since I don't have a better idea), but we should document that we're *not* testing that the queries are actually correct.  For instance, if someone swaps the write and flush delta in the query the tests won't be impacted.

Reply:
Fixed. I added some explanations and warning in comments.